package/native-package/src/optionalDependencies/takePhoto.ts

```diff
@@ -29,46 +32,63 @@ export const takePhoto = ImagePicker
       }
       try {
         const result = await ImagePicker.launchCamera({
+          mediaType: mediaType,
           quality: Math.min(Math.max(0, compressImageQuality), 1),
         });
-        if (!result.assets.length) {
+        if (!result.assets.length || result.didCancel) {
           return {
             cancelled: true,
           };
         }
-        const photo = result.assets[0];
-        if (photo.height && photo.width && photo.uri) {
-          let size: { height?: number; width?: number } = {};
-          if (Platform.OS === 'android') {
-            // Height and width returned by ImagePicker are incorrect on Android.
-            const getSize = (): Promise<{ height: number; width: number }> =>
-              new Promise((resolve) => {
-                Image.getSize(photo.uri, (width, height) => {
-                  resolve({ height, width });
+        const asset = result.assets[0];
+        if (asset.type.includes('video')) {
```

Comment:
Same here, but for `asset`

Reply:
!result.assets.length is checked on line 38

Reply:
But again, that guarantees nothing - if something goes wrong and we get something like `[undefined]` this will indeed fail. It will also fail if `result` doesn't contain `assets` too or if they're `null`.

Reply:
By the docs, that's unlikely to happen, but I can add a case for result.assets for safety as well.

Reply:
Done